### DISCUSSION TITLE
OpenTextDocument Without Activating other extensions

### DISCUSSION THREAD

**Discussion by Karelian-na** [microsoft/vscode-discussions#1192] [Extension Development QnA] [answered]
I'm now developing an extension, and i want to open a file using vscode api(because it provides native postion, range operation). Currently, i use vscode.openTextDocument, but when i open a cpp file, it will activate another extension (vscode-clangd), 
this is not what I want. I only use it to open a file and using range, postion api. so is there a way to achieve my goal?

**Comment by tamuratak** [comment, marked as answer]
[vscode-languageserver-textdocument](https://www.npmjs.com/package/vscode-languageserver-textdocument) would be useful. See the API:

- https://github.com/microsoft/vscode-languageserver-node/blob/main/textDocument/src/main.ts
